trin-history/src/events.rs

```diff
@@ -0,0 +1,94 @@
+use crate::network::HistoryNetwork;
+use discv5::TalkRequest;
+use log::{debug, error, warn};
+use rocksdb::DB;
+use std::sync::Arc;
+use tokio::sync::mpsc::UnboundedReceiver;
+use trin_core::portalnet::types::{
+    FindContent, FindNodes, FoundContent, Message, Nodes, Ping, Pong, Request, Response, SszEnr,
+};
+
+pub struct HistoryEvents {
+    pub network: HistoryNetwork,
+    pub db: Arc<DB>,
+    pub event_rx: UnboundedReceiver<TalkRequest>,
+}
+
+impl HistoryEvents {
```

Comment:
I'm a little nervous at how similar this and `StateEvents` are - besides some log messages, I don't notice any significant differences. Which means as we continue development, any change to one will likely need to be made on the other - and they could fall out of sync very quickly. Could/did you try to implement a common datatype that can be used in each subnetwork? If you want to handle that in a subsequent pr though, that's cool

Reply:
I moved the logic for handling the common requests (currently all requests are common) inside the overlay protocol, which probably makes perfect sense because every network implements the overlay protocol. This was actually the idea from the beginning but sometimes I get lost in refactoring :)